### DISCUSSION TITLE
Can you just import decompose in common source set?

### DISCUSSION THREAD

**Discussion by plusmobileapps** [badoo/Decompose#154] [Q&A]
I was looking over the Todo decompose sample on Jetbrains compose and noticed in the root module `build.gradle.kts` file that it's [exporting a specific version of decompose for the emulator or device architecture](https://github.com/JetBrains/compose-jb/blob/master/examples/todoapp/common/root/build.gradle.kts#L21). Is this actually needed, or is there some specific reason this is done for that project? The [installation instruction](https://arkivanov.github.io/Decompose/getting-started/) make it seem like the decompose dependency can be added to common source set only, or maybe there is some extra documentation I could add for Multiplatform setup? 

```kotlin
    ios {
        binaries {
            framework {
                when (val target = this.compilation.target.name) {
                    "iosX64" -> {
                        export(Deps.ArkIvanov.Decompose.decomposeIosX64)
                        export(Deps.ArkIvanov.MVIKotlin.mvikotlinMainIosX64)
                    }

                    "iosArm64" -> {
                        export(Deps.ArkIvanov.Decompose.decomposeIosArm64)
                        export(Deps.ArkIvanov.MVIKotlin.mvikotlinMainIosArm64)
                    }

                    else -> error("Unsupported target: $target")
                }
            }
        }
    }
```

**Comment by arkivanov** [comment]
Decompose has to be exported to the framework, so one can use things like `Value`, `RouterState`, and also to be able to instantiate the root component.

At the time I was writing it, the usual `export "com.arkivanov.decompose:decompose:<version>"` didn't work, and so I had to split by targets. Maybe with Kotlin 1.5.0 it will work.

We should definitely add this information to the docs, but before we should check how it will work with Kotlin 1.5.0.

Thanks for raising!

**Reply by Shabinder** [reply]
> At the time I was writing it, the usual `export "com.arkivanov.decompose:decompose:<version>"` didn't work, and so I had to split by targets. Maybe with Kotlin 1.5.0 it will work.


When I built my ios app, using **kotlin 1.4.32**, normal exporting using 
`export "com.arkivanov.decompose:decompose:<version>"`
did work

<img width="635" alt="image" src="https://user-images.githubusercontent.com/44765050/119633273-32cd7c80-be2f-11eb-9b40-810c6ca803b7.png">
Link:
https://github.com/Shabinder/SpotiFlyer/blob/72c3f1e9c54ea4a4e1cc5ac1519f27fcc2bd818d/common/root/build.gradle.kts#L24

**Reply by arkivanov** [reply]
That's sounds promising, at the time of writing the Todo app it didn't work, hence the code acrobatics. I will check once again the Todo app tonight, or you can check by yourself if you are eager. 🙃

**Reply by arkivanov** [reply]
Indeed, it seems we can now use common artifacts for iOS framework export. Opened a PR for TodoApp: https://github.com/JetBrains/compose-jb/pull/706.

**Reply by plusmobileapps** [reply]
Nice, thanks for all the answers and even updating the example! This should make the setup a lot easier 😅

**Reply by arkivanov** [reply]
The PR is merged, however according to documentation it shouldn't work) https://kotlinlang.org/docs/mpp-build-native-binaries.html#export-dependencies-to-binaries

**Reply by arkivanov** [reply]
![Screenshot_20210528-175950](https://user-images.githubusercontent.com/26204457/120018204-be394000-bfde-11eb-91b7-5b2adf38d0e3.png)
